packages/graphql/src/schema/resolvers/cypher.ts

```diff
@@ -153,13 +153,25 @@ export default function cypherResolver({
         }
 
         const initApocParamsStrs = ["auth: $auth", ...(context.cypherParams ? ["cypherParams: $cypherParams"] : [])];
-        const apocParams = Object.entries(resolveTree.args).reduce(
-            (r: { strs: string[]; params: any }, entry) => {
-                return {
-                    strs: [...r.strs, `${entry[0]}: $${entry[0]}`],
-                    params: { ...r.params, [entry[0]]: entry[1] },
-                };
-            },
+
+        // Null default argument values are not passed into the resolve tree therefore these are not being passed to
+        // `apocParams` below causing a runtime error when executing.
+        const nullArgumentValues = field.arguments.reduce(
+            (res, argument) => ({
+                ...res,
+                ...{ [argument.name.value]: null },
+            }),
+            {}
+        );
+
+        // Overwrite null arguments with those defined in resolve tree
+        const argumentsWithNullValues = { ...nullArgumentValues, ...resolveTree.args };
```

Comment:
I don't know why, but I find this variable name very confusing, it sounds like "all of the arguments in this object have a null value". Maybe just `arguments` if that's not already declared in this scope?

Reply:
Sorry, it does look confusing. Unfortunately, `arguments` is a reserved word in javascript and `args` is taken. Perhaps `apocArgs`? I do agree that it should be changed to be clearer.

It is only ever used in generating the `apocParams`. We could skip the entire problem and put it directly in `Object.entries` and let the comment above `nullArgumentValues` do the work.

Reply:
Of course it is, easy to forget all of the reserved words!

I think just pop it directly in `Object.entries` given that it's only spreading a couple of objects and not a huge multiline function. 🙂

Reply:
Fixed in https://github.com/neo4j/graphql/pull/769/commits/b27b37efc39a595019b8668b235737d14b7d6b65